2018/11/05/testingmlp.cpp

```diff
@@ -1,1688 +1,33 @@
-    val1 = bigarray[val1];
-    val2 = bigarray[val2];
-    val3 = bigarray[val3];
-    val4 = bigarray[val4];
-    val5 = bigarray[val5];
-    val6 = bigarray[val6];
-    val7 = bigarray[val7];
-    val8 = bigarray[val8];
-    val9 = bigarray[val9];
-    val10 = bigarray[val10];
-    val11 = bigarray[val11];
-    val12 = bigarray[val12];
-    val13 = bigarray[val13];
-    val14 = bigarray[val14];
-    val15 = bigarray[val15];
-    val16 = bigarray[val16];
-    val17 = bigarray[val17];
-    val18 = bigarray[val18];
-    val19 = bigarray[val19];
-    val20 = bigarray[val20];
-    val21 = bigarray[val21];
-    val22 = bigarray[val22];
-    val23 = bigarray[val23];
-    val24 = bigarray[val24];
-    val25 = bigarray[val25];
-    val26 = bigarray[val26];
-    val27 = bigarray[val27];
-    val28 = bigarray[val28];
-  }
-  return val1 + val2 + val3 + val4 + val5 + val6 + val7 + val8 + val9 + val10 +
-         val11 + val12 + val13 + val14 + val15 + val16 + val17 + val18 + val19 +
-         val20 + val21 + val22 + val23 + val24 + val25 + val26 + val27 + val28;
-}
+#include <random>
 
-float time_me28(uint64_t *bigarray, size_t howmanyhits, size_t repeat) {
-  clock_t begin_time, end_time;
-  float mintime = 99999999999;
-  uint64_t bogus = 0;
-  while (repeat-- > 0) {
-    begin_time = clock();
-    bogus += naked_access_28(bigarray, howmanyhits);
-    end_time = clock();
-    float tv = float(end_time - begin_time) / CLOCKS_PER_SEC;
-    if (tv < mintime)
-      mintime = tv;
-  }
-  if (bogus == 0x010101) {
-    printf("ping!");
-  }
-  std::cout << 28 << " " << mintime << std::endl;
-  return mintime;
+static uint64_t murmur64(uint64_t h) {
+  h ^= h >> 33;
+  h *= UINT64_C(0xff51afd7ed558ccd);
+  h ^= h >> 33;
+  h *= UINT64_C(0xc4ceb9fe1a85ec53);
+  h ^= h >> 33;
+  return h;
 }
 
-uint64_t naked_access_29(uint64_t *bigarray, size_t howmanyhits) {
-  uint64_t val1 = 1;
-  uint64_t val2 = 2;
-  uint64_t val3 = 3;
-  uint64_t val4 = 4;
-  uint64_t val5 = 5;
-  uint64_t val6 = 6;
-  uint64_t val7 = 7;
-  uint64_t val8 = 8;
-  uint64_t val9 = 9;
-  uint64_t val10 = 10;
-  uint64_t val11 = 11;
-  uint64_t val12 = 12;
-  uint64_t val13 = 13;
-  uint64_t val14 = 14;
-  uint64_t val15 = 15;
-  uint64_t val16 = 16;
-  uint64_t val17 = 17;
-  uint64_t val18 = 18;
-  uint64_t val19 = 19;
-  uint64_t val20 = 20;
-  uint64_t val21 = 21;
-  uint64_t val22 = 22;
-  uint64_t val23 = 23;
-  uint64_t val24 = 24;
-  uint64_t val25 = 25;
-  uint64_t val26 = 26;
-  uint64_t val27 = 27;
-  uint64_t val28 = 28;
-  uint64_t val29 = 29;
-  size_t howmanyhits_perlane = howmanyhits / 29;
+template <size_t howmanylanes>
+uint64_t access(uint64_t *bigarray, size_t length, size_t howmanyhits) {
+  size_t howmanyhits_perlane = howmanyhits / howmanylanes;
+  uint64_t lanesums[howmanylanes];
+  for (size_t i = 0; i < howmanylanes; i++)
+    lanesums[i] = 0;
   for (size_t counter = 0; counter < howmanyhits_perlane; counter++) {
```

Comment:
I didn't really change any code in this file, except what was needed to extract the generated code out - this is just github matching up the diff weirdly.